### PR TITLE
dotnet-script use local tool

### PR DESCRIPTION
**Pull request by IsaacCalligeros95**
This change adds support for using a locally installed version of dotnet-script. Execution uses in order:
* Check if dotnet-script is on path with `where/which`
* Fall back to the bundled version

**Deprecation warning**
image

Note logs include additional logging for debugging purposes these lines are not included in the final PR.
**dotnet-script tool installed on WSL Bash script**
image

**WSL container without dotnet-script on path**
image

**WSL executing dotnet-script file**
image

**dotnet-script tool installed on Windows .cmd on path**
image

**dotnet-script tool installed on Windows .exe on path**
image

**No dotnet-script tool installed on Windows  (bundled)**
image

### CODE REVIEW

File: source/Calamari.Tests/Fixtures/Integration/Scripting/CSharpScriptEngineFixture.cs [hunk 1]
```diff
@@ -21,7 +22,8 @@ public void DotnetScript_CSharpDecryptsVariables()
                 var variables = GetVariables();
                 variables.Add(ScriptVariables.UseDotnetScript, bool.TrueString);
                 File.WriteAllText(scriptFile.FilePath, "System.Console.WriteLine(OctopusParameters[\"mysecrect\"]);");
-                var result = ExecuteScript(new DotnetScriptExecutor(), scriptFile.FilePath, variables);
+                var commandLineRunner = new TestCommandLineRunner(new InMemoryLog(), new CalamariVariables());
+                var result = ExecuteScript(new DotnetScriptExecutor(commandLineRunner), scriptFile.FilePath, variables);
```
Comment (IsaacCalligeros95) [inline source/Calamari.Tests/Fixtures/Integration/Scripting/CSharpScriptEngineFixture.cs:26]: I tried testing this, including a bundled dotnet-script package, extracting that to a path, and adding that path to the environment variables. In the test context this was not retrieving the executable path on the `where/which` commands. I sunk a lot of time into this and couldn't work out why this wasn't working.

**Review by hnrkndrssn — APPROVED** [review]
LGTM 👍

File: source/Calamari.Tests/KubernetesFixtures/InstallTools.cs [hunk 2]
```diff
@@ -67,7 +67,8 @@ public async Task InstallTerraform()
 
                                                             await DownloadTerraform(fileName, client, tuple.data, destinationDirectoryName);
 
-                                                            var terraformExecutable = Directory.EnumerateFiles(destinationDirectoryName).FirstOrDefault();
+                                                            var terraformExecutable = Directory.EnumerateFiles(destinationDirectoryName)
```
Comment (IsaacCalligeros95) [inline source/Calamari.Tests/KubernetesFixtures/InstallTools.cs:70]: Fixes test issues

**Review by hnrkndrssn — APPROVED** [review]
LGTM 👍

File: source/Calamari.AzureScripting/Scripts/AzureContext.ps1 [hunk 3]
```diff
@@ -185,14 +185,14 @@ Execute-WithRetry{
                     Initialize-AzContext
                 }
                 elseif (Get-AzureRmModuleInstalled) {
-                    if($OctopusAzureRMIsDepreciated -like [Boolean]::FalseString) {
-                        Write-Warning "Azure Resource Manager powershell module has reached end-of-life; please authenticate using Azure CLI or the Az module, Octopus will prevent usage of the AzureRM module in 2024.3."
-                        Initialize-AzureRmContext    
-                    }
-                    else {
+                    if($OctopusAzureRMIsDepreciated -like [Boolean]::TrueString) {
```
Comment (IsaacCalligeros95) [inline source/Calamari.AzureScripting/Scripts/AzureContext.ps1:188]: @rain-on I've inverted the check here for the sake of the tests. I can't see this changing any other behaviour but just wanted to raise this incase.

**Pull request merged**